include/pdal/util/Extractor.hpp

```diff
@@ -240,4 +240,158 @@ class PDAL_DLL LeExtractor : public Extractor
     }
 };
 
+
+/// Stream wrapper for input of binary data that converts from little-endian
+/// to host ordering.
+class PDAL_DLL SwitchableExtractor : public Extractor
```

Comment:
Understood. Qfit was already doing some goofy switching logic in its reader, which I tried to encapsulate with this `SwitchableExtractor`. I don't *think* this should introduce any performance hits that weren't already present in the qfit reader (though of course I could be wrong there).